src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1623 @@
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode0;
+
+        /// <summary>
+        /// The default I2C address
+        /// </summary>
+        public const byte I2cDefaultAddress = 0x24;
+
+        #endregion        
+
+        /// <summary>
+        /// Create a PN532 using Serial Port
+        /// </summary>
+        /// <param name="portName">The port name</param>
+        public Pn532(string portName)
+        {
+#if DEBUG
+            LogLevel = LogLevel.Debug;
+#endif
+
+            // Data bit : 8 bits,
+            // Parity bit : none,
+            // Stop bit : 1 bit,
+            // Baud rate : 115 200 bauds,
+            LogInfo.Log("Opening serial port 115200, Parity.None, 8 bits, StopBits.One", LogLevel.Debug);
+            _serialPort = new SerialPort(portName, 115200, Parity.None, 8, StopBits.One);
+            _serialPort.ReadTimeout = SerialReadTimeOut;
+            _serialPort.WriteTimeout = SerialReadTimeOut;
+            _serialPort.Open();
+            // Setting up internals as default version
+            _parametersFlags = ParametersFlags.fAutomaticATR_RES | ParametersFlags.fAutomaticRATS | ParametersFlags.fISO14443_4_PICC;
+            _securityAccessModuleMode = SecurityAccessModuleMode.Normal;
+            _virtualCardTimeout = 20;
+
+            WakeUp();
+            // Set the SAM
+            var ret = SetSecurityAccessModule();
+            //var ret = SetSAM();
+            LogInfo.Log($"Setting SAM changed: {ret}", LogLevel.Info);
+            // Check the version
+            if (!CheckVersion())
+                throw new Exception("Can't find a PN532");
+            // Apply default parameters
+            ret = SetParameters(ParametersFlags.fAutomaticATR_RES | ParametersFlags.fAutomaticRATS);
+            LogInfo.Log($"Setting Parameters Falgs changed: {ret}", LogLevel.Info);
+        }
+
+        /// <summary>
+        /// Create a PN532 using SPI
+        /// </summary>
+        /// <param name="spiDevice"></param>
+        public Pn532(SpiDevice spiDevice)
+        {
+#if DEBUG
+            LogLevel = LogLevel.Debug;
+#endif
+
+            _spiDevice = spiDevice;
+            _controller = new GpioController();
+            _controller.OpenPin(_pin, PinMode.Output);
+            _controller.Write(_pin, PinValue.High);
+            Thread.Sleep(2);
+            LogLevel = LogLevel.Debug;
+            WakeUp();
+            var ret = SetSecurityAccessModule();
+            //var ret = SetSAM();
+            LogInfo.Log($"Setting SAM changed: {ret}", LogLevel.Info);
+            // Check the version
+            if (!CheckVersion())
+                throw new Exception("Can't find a PN532");
+            // Apply default parameters
+            ret = SetParameters(ParametersFlags.fAutomaticATR_RES | ParametersFlags.fAutomaticRATS);
+            LogInfo.Log($"Setting Parameters Falgs changed: {ret}", LogLevel.Info);
+            ret = SetSecurityAccessModule();
```

Comment:
why does this need to be set twice?

Reply:
The first one always return a timeout or equivalent. It's the case right after waking it up. So it needs to be reapplied after. Will clean the comment and add this.